File: backend/core/ontology.py
COMPONENTS = ["Stat", "Alg", "Vis", "Int"]
MEASURES = ["AC", "PD", "Ct"]
LEVELS = ["High", "Low"]

ALL_ENTITY_IDS = [f"{c}-{l}-{m}" for c in COMPONENTS for m in MEASURES for l in LEVELS]

ENTITY_DESCRIPTIONS = {
    "Stat-High-AC": "Aggressive statistical aggregation", "Stat-Low-AC": "No statistical aggregation",
    "Stat-High-PD": "Statistics introduce errors", "Stat-Low-PD": "Statistics are accurate",
    "Stat-High-Ct": "Statistics are expensive", "Stat-Low-Ct": "Statistics are cheap",
    "Alg-High-AC": "Aggressive algorithmic filtering", "Alg-Low-AC": "No algorithmic filtering",
    "Alg-High-PD": "Algorithm produces errors", "Alg-Low-PD": "Algorithm is accurate",
    "Alg-High-Ct": "Algorithm is expensive", "Alg-Low-Ct": "Algorithm is cheap",
    "Vis-High-AC": "Summary visualization", "Vis-Low-AC": "Raw data visualization — cluttered",
    "Vis-High-PD": "Visualization misleads", "Vis-Low-PD": "Visualization is accurate",
    "Vis-High-Ct": "High cognitive load", "Vis-Low-Ct": "Easy to read",
    "Int-High-AC": "Interaction hides detail", "Int-Low-AC": "Interaction reveals detail",
    "Int-High-PD": "Interaction causes errors", "Int-Low-PD": "Interaction is precise",
    "Int-High-Ct": "Too many clicks, complex", "Int-Low-Ct": "Quick and effortless",
}

NEVER_SYMPTOMS = [f"{c}-Low-{m}" for c in COMPONENTS for m in ["PD", "Ct"]]

def get_entity_info(entity_id):
    parts = entity_id.split("-")
    return {"id": entity_id, "component": parts[0], "level": parts[1], "measure": parts[2],
            "description": ENTITY_DESCRIPTIONS.get(entity_id, ""), "component_color": "#5C6B4F"}

def get_all_entities():
    return [get_entity_info(e) for e in ALL_ENTITY_IDS]

File: react-app/src/data/ontology.js
export const COMPONENTS = {
  Stat: { name: "Statistics", color: "#5C6B4F", light: "#e8ede5" },
  Alg:  { name: "Algorithm",  color: "#4A4A4A", light: "#e8e8e8" },
  Vis:  { name: "Visualization", color: "#7A6652", light: "#ede8e2" },
  Int:  { name: "Interaction",   color: "#6B6B6B", light: "#ebebeb" },
};
export const ENTITY_DESC = {
  "Stat-High-AC": "Aggressive statistical aggregation — heavy averaging, binning",
  "Stat-Low-AC":  "No statistical aggregation — raw data passes through",
  "Stat-High-PD": "Aggregation introduces errors — hides patterns/outliers",
  "Stat-Low-PD":  "Statistics are accurate and faithful",
  "Stat-High-Ct": "Statistical processing is expensive/slow",
  "Stat-Low-Ct":  "Statistical processing is fast and cheap",
  "Alg-High-AC":  "Aggressive filtering/clustering/ML — removes much detail",
  "Alg-Low-AC":   "Minimal algorithmic filtering — data passes through",
  "Alg-High-PD":  "Algorithm errors — wrong clusters, missed anomalies",
  "Alg-Low-PD":   "Algorithm is accurate and reliable",
  "Alg-High-Ct":  "Algorithm is slow or resource-intensive",
  "Alg-Low-Ct":   "Algorithm is fast and lightweight",
  "Vis-High-AC":  "Summary visualization — overview, glyphs, aggregated charts",
  "Vis-Low-AC":   "Raw data visualization — cluttered, many marks on screen",
  "Vis-High-PD":  "Visualization misleads — bad colors, occlusion, deceptive scales",
  "Vis-Low-PD":   "Visualization is accurate and readable",
  "Vis-High-Ct":  "Visualization has high cognitive load — slow to comprehend",
  "Vis-Low-Ct":   "Visualization is intuitive and quick to read",
  "Int-High-AC":  "Interaction hides detail — collapsed menus, heavy filtering",
  "Int-Low-AC":   "Interaction reveals detail — drill-down, details-on-demand",
  "Int-High-PD":  "Interaction causes errors — confusing controls, wrong selections",
  "Int-Low-PD":   "Interaction is precise and reliable",
  "Int-High-Ct":  "Too many clicks, slow response, complex multi-step workflow",
  "Int-Low-Ct":   "Interactions are quick and effortless",
};
export const STAGES = {
  symptom:     { color: "#B91C1C", bg: "#FEF2F2", border: "#FECACA", label: "Symptom",     icon: "S" },
  cause:       { color: "#1D4ED8", bg: "#EFF6FF", border: "#BFDBFE", label: "Cause",       icon: "C" },
  remedy:      { color: "#047857", bg: "#ECFDF5", border: "#A7F3D0", label: "Remedy",      icon: "R" },
  side_effect: { color: "#D97706", bg: "#FFFBEB", border: "#FDE68A", label: "Side-Effect", icon: "!" },
};
export const getComp = (id) => id.split("-")[0];
export const getCompColor = (id) => COMPONENTS[getComp(id)]?.color || "#666";
